src/main/java/org/xstefank/check/TitleCheck.java

```diff
@@ -2,27 +2,33 @@
 
 import com.fasterxml.jackson.databind.JsonNode;
 import org.xstefank.model.Utils;
-
+import org.xstefank.model.yaml.RegexDefinition;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class TitleCheck implements Check {
 
-    private static final String ERROR_MESSAGE = "Invalid title format";
+    static final String DEFAULT_MESSAGE = "Invalid title content";
 
     private Pattern pattern;
+    private String message;
 
-    public TitleCheck(Pattern pattern) {
-        this.pattern = pattern;
+    public TitleCheck(RegexDefinition title) {
+        if (title == null || title.getPattern() == null) {
+            throw new IllegalArgumentException("Input argument cannot be null!");
+        } else {
```

Comment:
this else block is also unnecessary ;) please try to always check comments also for other occurrences